### PR TITLE
junit5 upgrade. add docs. add initial report handling of @BeforeAll

### PR DESCRIPTION
**Pull request by MykolaGolubyev**

### CODE REVIEW

**Comment by MykolaGolubyev** [comment]
Asked about junit5 potential way of handling `@BeforeAll` in a better way https://github.com/junit-team/junit5/issues/1769

File: webtau-junit5-examples/src/test/java/com/example/tests/junit5/CustomerCrudSeparatedJavaTest.java [hunk 1]
```diff
@@ -1,61 +1,41 @@
...
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package com.example.tests.rest;
+package com.example.tests.junit5;
```
Comment (tsiq-karold) [inline webtau-junit5-examples/src/test/java/com/example/tests/junit5/CustomerCrudSeparatedJavaTest.java]: Why did you delete the copyright notice?

File: pom.xml [hunk 2]
```diff
@@ -433,6 +434,12 @@
                     <version>3.1.0</version>
                 </plugin>
 
+                <plugin>
+                    <groupId>org.apache.maven.plugins</groupId>
+                    <artifactId>maven-surefire-plugin</artifactId>
+                    <version>3.0.0-M3</version>
```
[the remark this answers is not in the document]
Comment (MykolaGolubyev) [reply pom.xml]: That's the one on maven site :( and I think the one that supports junit 5

File: pom.xml [hunk 3]
```diff
@@ -613,7 +620,8 @@
                                     <goal>groovydoc</goal>
                                 </goals>
                                 <configuration>
-                                    <groovydocOutputDirectory>${project.build.directory}/apidocs</groovydocOutputDirectory>
+                                    <groovydocOutputDirectory>${project.build.directory}/apidocs
```
[the remark this answers is not in the document]
Comment (MykolaGolubyev) [reply pom.xml]: ouch. didn't see it.

File: webtau-junit5-examples/src/test/java/com/example/tests/junit5/CustomerCrudSeparatedJavaTest.java (hunk 1 above)
Comment (MykolaGolubyev) [reply webtau-junit5-examples/src/test/java/com/example/tests/junit5/CustomerCrudSeparatedJavaTest.java]: it is being pulled into a doc as an example

File: webtau-docs/webtau/generic-runners/JUnit-4.md [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # WebTau Runner
 
-You can use webtau `http.` and `browser.` methods as in a junit test, but to enable reporting you need to use 
+You can use webtau `http.` and `browser.` methods as in a junit 4 tests, but to enable reporting you need to use
```
[the remark this answers is not in the document]
Comment (MykolaGolubyev) [reply webtau-docs/webtau/generic-runners/JUnit-4.md]: `as is`
